org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/NewCUProposal.java

```diff
@@ -391,22 +390,32 @@ private String constructCUContent(ICompilationUnit cu, String typeContent, Strin
 	 * Called from createType to construct the source for this type
 	 */
 	private String constructTypeStub(ICompilationUnit parentCU, String name, int modifiers, String lineDelimiter) throws CoreException {
-		StringBuffer buf = new StringBuffer();
+		StringBuilder buf = new StringBuilder();
 
 		buf.append(Flags.toString(modifiers));
 		if (modifiers != 0) {
 			buf.append(' ');
 		}
+
+		String[] permittedNames = fCompilationUnit.findPrimaryType().getPermittedSubtypeNames();
```

Comment:
Assign fCompilationUnit.findPrimaryType() to a variable and reuse it on L413 and L434